<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ProtocolInfoForm</class>
 <widget class="QWidget" name="ProtocolInfoForm">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>672</width>
    <height>626</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QFormLayout" name="formLayout">
   <item row="0" column="0">
    <widget class="QLabel" name="label_EntryType">
     <property name="text">
      <string>Entry type:</string>
     </property>
    </widget>
   </item>
   <item row="0" column="1">
    <widget class="QLabel" name="label_EntryTypeContent">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
   </item>
   <item row="3" column="0">
    <widget class="QLabel" name="label_ProtocolName">
     <property name="text">
      <string>Protocol:</string>
     </property>
    </widget>
   </item>
   <item row="3" column="1">
    <widget class="QLineEdit" name="lineEdit_ProtocolName">
     <property name="toolTip">
      <string>An array of characters that contains a human-readable name identifying the protocol, for example &quot;MSAFD Tcpip [UDP/IP]&quot;. The maximum number of characters allowed is WSAPROTOCOL_LEN, which is defined to be 255.</string>
     </property>
    </widget>
   </item>
   <item row="4" column="0">
    <widget class="QLabel" name="label_ProviderId">
     <property name="text">
      <string>Provider ID:</string>
     </property>
    </widget>
   </item>
   <item row="4" column="1">
    <widget class="QLineEdit" name="lineEdit_ProviderId">
     <property name="enabled">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="5" column="0">
    <widget class="QLabel" name="label_CatalogEntryId">
     <property name="text">
      <string>Catalog Entry ID:</string>
     </property>
    </widget>
   </item>
   <item row="5" column="1">
    <widget class="QLabel" name="label_CatalogEntryIdValue">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
   </item>
   <item row="6" column="0">
    <widget class="QLabel" name="label_Version">
     <property name="text">
      <string>Version:</string>
     </property>
    </widget>
   </item>
   <item row="6" column="1">
    <widget class="QSpinBox" name="spinBox_Version">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="toolTip">
      <string>The protocol version identifier.</string>
     </property>
     <property name="maximum">
      <number>2</number>
     </property>
     <property name="value">
      <number>2</number>
     </property>
    </widget>
   </item>
   <item row="7" column="0">
    <widget class="QLabel" name="label_AddressFamily">
     <property name="text">
      <string>Address Family:</string>
     </property>
    </widget>
   </item>
   <item row="7" column="1">
    <widget class="QSpinBox" name="spinBox_AddressFamily">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="maximum">
      <number>35</number>
     </property>
    </widget>
   </item>
   <item row="8" column="0">
    <widget class="QLabel" name="label_AddressLengthMax">
     <property name="text">
      <string>Max Socket Address Length:</string>
     </property>
    </widget>
   </item>
   <item row="8" column="1">
    <widget class="QSpinBox" name="spinBox_AddressLengthMax">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
    </widget>
   </item>
   <item row="9" column="0">
    <widget class="QLabel" name="label_AddressLengthMin">
     <property name="text">
      <string>Min Socket Address Length:</string>
     </property>
    </widget>
   </item>
   <item row="9" column="1">
    <widget class="QSpinBox" name="spinBox_AddressLengthMin">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
    </widget>
   </item>
   <item row="10" column="0">
    <widget class="QLabel" name="label_SocketType">
     <property name="text">
      <string>Socket Type:</string>
     </property>
    </widget>
   </item>
   <item row="10" column="1">
    <widget class="QSpinBox" name="spinBox_SocketType">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimum">
      <number>0</number>
     </property>
     <property name="maximum">
      <number>5</number>
     </property>
     <property name="value">
      <number>0</number>
     </property>
    </widget>
   </item>
   <item row="11" column="0">
    <widget class="QLabel" name="label_SocketProtocol">
     <property name="text">
      <string>Socket Protocol:</string>
     </property>
    </widget>
   </item>
   <item row="11" column="1">
    <widget class="QSpinBox" name="spinBox_SocketProtocol">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="maximum">
      <number>261</number>
     </property>
    </widget>
   </item>
   <item row="12" column="0">
    <widget class="QLabel" name="label_MaxOffset">
     <property name="text">
      <string>Socket Protocol Max Offset:</string>
     </property>
    </widget>
   </item>
   <item row="12" column="1">
    <widget class="QSpinBox" name="spinBox_MaxOffset">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="toolTip">
      <string>The maximum value that may be added to iProtocol when supplying a value for the protocol parameter to socket or WSASocket function. Not all protocols allow a range of values. When this is the case iProtocolMaxOffset is zero.</string>
     </property>
     <property name="maximum">
      <number>999999999</number>
     </property>
    </widget>
   </item>
   <item row="13" column="0">
    <widget class="QLabel" name="label_NetworkByteOrder">
     <property name="text">
      <string>Network Byte Order:</string>
     </property>
    </widget>
   </item>
   <item row="13" column="1">
    <widget class="QSpinBox" name="spinBox_NetworkByteOrder">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="toolTip">
      <string>Currently these values are manifest constants (BIGENDIAN and LITTLEENDIAN) that indicate either big-endian or little-endian with the values 0 and 1 respectively.</string>
     </property>
     <property name="maximum">
      <number>1</number>
     </property>
    </widget>
   </item>
   <item row="14" column="0">
    <widget class="QLabel" name="label_SecurityScheme">
     <property name="text">
      <string>Security Scheme:</string>
     </property>
    </widget>
   </item>
   <item row="14" column="1">
    <widget class="QSpinBox" name="spinBox_SecurityScheme">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="toolTip">
      <string>The type of security scheme employed (if any). A value of SECURITY_PROTOCOL_NONE (0) is used for protocols that do not incorporate security provisions.</string>
     </property>
    </widget>
   </item>
   <item row="15" column="0">
    <widget class="QLabel" name="label_MaxMessageSize">
     <property name="text">
      <string>Max Message Size:</string>
     </property>
    </widget>
   </item>
   <item row="15" column="1">
    <widget class="QSpinBox" name="spinBox_MaxMessageSize">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="toolTip">
      <string>The maximum message size, in bytes, supported by the protocol. This is the maximum size that can be sent from any of the host's local interfaces. For protocols that do not support message framing, the actual maximum that can be sent to a given address may be less. There is no standard provision to determine the maximum inbound message size. The following special values are defined.</string>
     </property>
     <property name="minimum">
      <number>-2147483647</number>
     </property>
     <property name="maximum">
      <number>2147483647</number>
     </property>
    </widget>
   </item>
   <item row="16" column="0">
    <widget class="QLabel" name="label_ServiceFlags1">
     <property name="text">
      <string>ServiceFlags1:</string>
     </property>
    </widget>
   </item>
   <item row="16" column="1">
    <widget class="QSpinBox" name="spinBox_ServiceFlags1">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="prefix">
      <string>0x</string>
     </property>
     <property name="minimum">
      <number>0</number>
     </property>
     <property name="maximum">
      <number>1048575</number>
     </property>
     <property name="stepType">
      <enum>QAbstractSpinBox::DefaultStepType</enum>
     </property>
     <property name="value">
      <number>0</number>
     </property>
     <property name="displayIntegerBase">
      <number>16</number>
     </property>
    </widget>
   </item>
   <item row="17" column="0">
    <widget class="QLabel" name="label_ServiceFlags2">
     <property name="text">
      <string>ServiceFlags2:</string>
     </property>
    </widget>
   </item>
   <item row="17" column="1">
    <widget class="QSpinBox" name="spinBox_ServiceFlags2">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="prefix">
      <string>0x</string>
     </property>
     <property name="maximum">
      <number>0</number>
     </property>
     <property name="displayIntegerBase">
      <number>16</number>
     </property>
    </widget>
   </item>
   <item row="18" column="0">
    <widget class="QLabel" name="label_ServiceFlags3">
     <property name="text">
      <string>ServiceFlags3:</string>
     </property>
    </widget>
   </item>
   <item row="18" column="1">
    <widget class="QSpinBox" name="spinBox_ServiceFlags3">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="prefix">
      <string>0x</string>
     </property>
     <property name="maximum">
      <number>0</number>
     </property>
     <property name="displayIntegerBase">
      <number>16</number>
     </property>
    </widget>
   </item>
   <item row="19" column="0">
    <widget class="QLabel" name="label_ServiceFlags4">
     <property name="text">
      <string>ServiceFlags4:</string>
     </property>
    </widget>
   </item>
   <item row="19" column="1">
    <widget class="QSpinBox" name="spinBox_ServiceFlags4">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="prefix">
      <string>0x</string>
     </property>
     <property name="maximum">
      <number>0</number>
     </property>
     <property name="displayIntegerBase">
      <number>16</number>
     </property>
    </widget>
   </item>
   <item row="20" column="0">
    <widget class="QLabel" name="label_ProviderFlags">
     <property name="text">
      <string>ProviderFlags:</string>
     </property>
    </widget>
   </item>
   <item row="20" column="1">
    <widget class="QSpinBox" name="spinBox_ProviderFlags">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="prefix">
      <string>0x</string>
     </property>
     <property name="maximum">
      <number>31</number>
     </property>
     <property name="value">
      <number>0</number>
     </property>
     <property name="displayIntegerBase">
      <number>16</number>
     </property>
    </widget>
   </item>
   <item row="22" column="0">
    <widget class="QLabel" name="label_ProtocolChainLength">
     <property name="text">
      <string>Protocol Chain Length:</string>
     </property>
    </widget>
   </item>
   <item row="22" column="1">
    <widget class="QSpinBox" name="spinBox_ProtocolChainLength">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="prefix">
      <string/>
     </property>
     <property name="maximum">
      <number>7</number>
     </property>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QLabel" name="label_Dll32Path">
     <property name="text">
      <string>DLL32 Path:</string>
     </property>
    </widget>
   </item>
   <item row="1" column="1">
    <widget class="QLineEdit" name="lineEdit_Dll32Path">
     <property name="enabled">
      <bool>false</bool>
     </property>
    </widget>
   </item>
   <item row="23" column="0">
    <widget class="QLabel" name="label_ProtocolChain">
     <property name="text">
      <string>Protocol Chain:</string>
     </property>
    </widget>
   </item>
   <item row="23" column="1">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QSpinBox" name="spinBox_ProtocolChain_1">
       <property name="maximum">
        <number>999999</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="spinBox_ProtocolChain_2">
       <property name="maximum">
        <number>999999</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="spinBox_ProtocolChain_3">
       <property name="maximum">
        <number>999999</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="spinBox_ProtocolChain_4">
       <property name="maximum">
        <number>999999</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="spinBox_ProtocolChain_5">
       <property name="maximum">
        <number>999999</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="spinBox_ProtocolChain_6">
       <property name="maximum">
        <number>999999</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="spinBox_ProtocolChain_7">
       <property name="maximum">
        <number>999999</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="2" column="0">
    <widget class="QLabel" name="label_Dll64Path">
     <property name="text">
      <string>DLL64 Path:</string>
     </property>
    </widget>
   </item>
   <item row="2" column="1">
    <widget class="QLineEdit" name="lineEdit_Dll64Path">
     <property name="enabled">
      <bool>false</bool>
     </property>
    </widget>
   </item>
   <item row="21" column="0">
    <widget class="QLabel" name="label_ProviderReserved">
     <property name="text">
      <string>Provider Reserved:</string>
     </property>
    </widget>
   </item>
   <item row="21" column="1">
    <widget class="QSpinBox" name="spinBox_ProviderReserved">
     <property name="suffix">
      <string/>
     </property>
     <property name="prefix">
      <string>0x</string>
     </property>
     <property name="minimum">
      <number>-2147483647</number>
     </property>
     <property name="maximum">
      <number>2147483647</number>
     </property>
     <property name="stepType">
      <enum>QAbstractSpinBox::DefaultStepType</enum>
     </property>
     <property name="value">
      <number>0</number>
     </property>
     <property name="displayIntegerBase">
      <number>16</number>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
